plugins/package-managers/bower/src/main/kotlin/Bower.kt

```diff
@@ -227,7 +227,7 @@ private fun parseDependencyTree(
         // about the subtree rooted at the parent from that other node containing the full dependency
         // information.
         // See https://github.com/bower/bower/blob/6bc778d/lib/core/Manager.js#L557 and below.
-        val alternativeNode = checkNotNull(alternativeNodes[info.key])
+        val alternativeNode = alternativeNodes.getValue(info.key!!)
```

Comment:
This triggers an inspection hint about the use of `!!`. I wonder whether we can work around that by making `key` non-nullable, e.g. by ensuring to not call it on root projects?

Reply:
Is it just me or have the warnings about `!!` become a bit too much. 

I just wanted to make obvious what previously was a bit hidden, but this warning blocks the change right?
(If so, is it possible to go back to the previous way of flagging `!!` ?)

Reply:
Yes, this blocks the merge. But previously `!!` was not flagged at all, I believe. I prefer the current way of flagging it, and if you're *really* sure `!!` is the right thing to do in a specific case, it should be whitelisted via `@Suppress("UnsafeCallOnNullableType")`.

Reply:
Can we revisit the decision whether we want to flag all uses of `!!` really? 
I've spent quite some time working around cases, where `!!` was perfectly fine. Huge waste of time imo. 
If there is a convincing rationale why `!!` should be forbidden, I may change my mind. But I'm not aware of any, in particular
1. `!!` seems forbidden while `checkNotNull()` seems fine
2. Whether `!!` is ok is a case by case decision, and I'm pretty sure all approvers in our community already 
    pay good attention to any ocurence of `!!`, so flagging is not needed as we handle it already.

Reply:
I went for the `supress`, because the observation that the previous `checkNotNull` never failed implies that no lockup with a `null` key has been done.

Reply:
> Can we revisit the decision whether we want to flag all uses of `!!` really?

Sure, feel free to bring it up in a Kotlin dev meeting (not in the next three weeks, though, as both @mnonnenmacher and I will be on vacation).